client/src/game/ui/notes.ts

```diff
@@ -219,7 +219,10 @@ export function setAllCardIndicators(): void {
   }
 }
 
-function stripHTMLTags(input: string) {
-  const doc = new DOMParser().parseFromString(input, "text/html");
-  return doc.body.textContent ?? "";
+function convertHTMLEntities(input: string) {
+  return String(input)
+    .replace(/&/g, "&amp;")
+    .replace(/</g, "&lt;")
+    .replace(/>/g, "&gt;")
+    .replace(/"/g, "&quot;");
```

Comment:
If you want to make this really robust it might make sense just to try putting the contents inside of a `p` and see how to browser escapes it. I tested that this works on Firefox.
```suggestion
function convertHTMLEntities(input: string) {
  const p = document.createElement("p");
  p.appendChild(document.createTextNode(input))
  return p.innerHTML
```

At that point I wonder why we don't just remove all this logic and do it where we're displaying the notes instead

Reply:
is that safe

Reply:
why wouldn't it be? It doesn't interpret the text passed to `createTextNode` as HTML

Reply:
It only converts &, <, >

Reply:
https://stackoverflow.com/questions/6234773/can-i-escape-html-special-chars-in-javascript

anax is this where you found the code?

Reply:
its probably safer to use jquery's test function.
its an antipattern to roll your own security

Reply:
> https://stackoverflow.com/questions/6234773/can-i-escape-html-special-chars-in-javascript
> 
> anax is this where you found the code?

Nope, just wrote it

Reply:
> its probably safer to use jquery's text function.
> its an antipattern to roll your own security

Can't use text() on an input element

Reply:
can't you use it as a replacement for the function that you hand-crafted?

Reply:
I can create a new element, paste the input's value and get it back via the text() function, is this ok?

Reply:
oh, i guess that is pretty similar to what tim is proposing

Reply:
So, use this?

function convertHTMLEntities(input: string) {
  const p = document.createElement("p");
  p.appendChild(document.createTextNode(input))
  return p.innerHTML

Reply:
ok i looked it up and jquery's text() function just does the code that tim suggested, so let's go with tim's code instead

Reply:
also, i am talking to you on discord, but you are not online =p